public/components/metrics/index.tsx

```diff
@@ -82,6 +86,11 @@ export const Home = ({
       dispatch(setSelectedDataSourceMDSId(id));
     }, 300);
   };
+  const dataSourceFilterFn = (dataSource: SavedObject<DataSourceAttributes>) => {
+    const dataSourceVersion = dataSource?.attributes?.dataSourceVersion || '';
+    return semver.satisfies(dataSourceVersion, pluginManifest.supportedOSDataSourceVersions);
+  };
```

Comment:
instead of creating the same function in each component, can it be declared in some utils file and imported?